### ISSUE TITLE
Cannot use graphql Schema from a submodule

### ISSUE THREAD

**Issue by Sceat** [apollographql/apollo-server#3773]
## Error
```
Error: Cannot use GraphQLSchema "[object GraphQLSchema]" from another module or realm.

Ensure that there is only one instance of "graphql" in the node_modules
directory. If different versions of "graphql" are the dependencies of other
relied on modules, use "resolutions" to ensure only one version is installed.

https://yarnpkg.com/en/docs/selective-version-resolutions

Duplicate "graphql" modules cannot be used at the same time since different
versions may have different capabilities and behavior. The data from one
version used in the function from another could produce confusing and
spurious results.
```

## How
When working in a monorepo and importing a graphql schema from a submodule made with `makeExecutableSchema` in another submodule

```
.
├── ...
├── packages                  
│   ├── bar              
│   │	└── package.json
│	│			# "apollo-server": "^2.10.0"       
│   ├── foo              
│   │	└── package.json
│	│			# "apollo-server": "^2.10.0"
│	│			# "foo": "file:../foo"	           
│   └── ...                             
└── ...
```

repository: https://github.com/Sceat/graphql-dependency-duplication

## Why
I guess `@types/graphql-upload` is breaking submodules by forcing a Graphql dependency, it should be a dev dependency on `apollo-server-core`, or DefinitelyTyped [should use `graphql` as a peerDependency](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/graphql-upload/package.json#L4)

**Comment by Sceat** [comment]
as a quick workaround we can install graphql globally and link in every submodule
```
npm i -g graphql
npm link # in /bar and /foo
```

**Comment by ChinW** [comment]
I meet the same issue when using webpack

```
"dependencies": {
    "apollo-server": "^2.18.2",
    "babel-loader": "^8.1.0",
    "graphql": "^15.3.0",
    "node-loader": "^1.0.0",
    "webpack": "^5.0.0",
    "webpack-node-externals": "^2.5.0"
  },
```

```
//webpack
mode: "development",
  target: "node",
  module: {
    rules: [
      {
        test: /\.(j|t)sx?$/,
        exclude: /node_modules/,
        use: {
          loader: "babel-loader",
          options: {
            cacheDirectory: false
          },
        },
      }
    ],
  },
  resolve: {
    extensions: [".ts", ".tsx", ".js", ".jsx"],
  },
```

npm link graphql doesnt work for me, 

```
bash > npm ls graphql
├─┬ apollo-server@2.18.2
│ └─┬ apollo-server-core@2.18.2
│   └─┬ @types/graphql-upload@8.0.4
│     └── graphql@15.3.0  deduped
└── graphql@15.3.0 
```

**Comment by ChinW** [comment]
I meet the same issue when using webpack

```
"dependencies": {
    "apollo-server": "^2.18.2",
    "babel-loader": "^8.1.0",
    "graphql": "^15.3.0",
    "node-loader": "^1.0.0",
    "webpack": "^5.0.0",
    "webpack-node-externals": "^2.5.0"
  },
```

```
//webpack
mode: "development",
  target: "node",
  module: {
    rules: [
      {
        test: /\.(j|t)sx?$/,
        exclude: /node_modules/,
        use: {
          loader: "babel-loader",
          options: {
            cacheDirectory: false
          },
        },
      }
    ],
  },
  resolve: {
    extensions: [".ts", ".tsx", ".js", ".jsx"],
  },
```

npm link graphql doesnt work for me, 
a
```
bash > npm ls graphql
├─┬ apollo-server@2.18.2
│ └─┬ apollo-server-core@2.18.2
│   └─┬ @types/graphql-upload@8.0.4
│     └── graphql@15.3.0  deduped
└── graphql@15.3.0 
```

**Comment by glasser** [comment]
This seems to have been fixed as described in #4637. In addition, `@types/graphql-upload` has not been a dependency of Apollo Server since v2.21.0.
